--- a/src/crypto/crypto.cc
+++ b/src/crypto/crypto.cc
@@ -257,8 +257,12 @@
   int pt_len = body_len - 16;
 
   if ( pt_len < 0 ) { /* super-assertion that pt_len does not equal AE_INVALID */
+#ifdef AMBER_MOSH_EMBEDDED
+    throw CryptoException("Invalid decrypted payload");
+#else
     fprintf( stderr, "BUG.\n" );
     exit( 1 );
+#endif
   }
 
   assert( (size_t)body_len <= ciphertext_buffer.len() );
